src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsOptionsModal.ts

```diff
@@ -0,0 +1,149 @@
+			localize("viewOptionsModal.title", "Configure View"),
+			'ViewOptionsModal',
+			telemetryService,
+			layoutService,
+			clipboardService,
+			themeService,
+			logService,
+			textResourcePropertiesService,
+			contextKeyService,
+			{ hasErrors: true, hasSpinner: true }
+		);
+	}
+
+	protected renderBody(container: HTMLElement): void {
+		const formWrapper = DOM.$<HTMLDivElement>('div#view-options-form');
+		formWrapper.style.padding = '10px';
+
+		DOM.append(container, formWrapper);
+
+		this._viewNameInput = this.createNameInput(formWrapper);
+
+	}
+
+	protected layout(height: number): void {
+
+	}
+
+	protected createNameInput(container: HTMLElement): InputBox {
+		return this.createInputBoxHelper(container, localize('viewOptionsModal.name', "View Name"), this._view.name, {
+			validationOptions: {
+				validation: (value: string) => {
+					if (!value) {
+						return ({ type: MessageType.ERROR, content: localize('viewOptionsModal.missingRequireField', "This field is required.") });
+					}
+					if (this._view.name !== value && !this._view.nameAvailable(value)) {
+						return ({ type: MessageType.ERROR, content: localize('viewOptionsModal.nameTaken', "This view name has already been taken.") });
+					}
+					return undefined;
+				}
+			},
+			ariaLabel: localize('viewOptionsModal.name', "View Name")
+		});
+	}
+
+	private createInputBoxHelper(container: HTMLElement, label: string, defaultValue: string = '', options?: IInputOptions): InputBox {
+		const inputContainer = DOM.append(container, DOM.$('.dialog-input-section'));
+		DOM.append(inputContainer, DOM.$('.dialog-label')).innerText = label;
+		const input = new InputBox(DOM.append(inputContainer, DOM.$('.dialog-input')), this._contextViewService, options);
+		input.value = defaultValue;
+		return input;
+	}
+
+	public override render() {
+		super.render();
+
+		this._submitButton = this.addFooterButton(localize('save', "Save"), () => this.onSubmitHandler());
+		this._cancelButton = this.addFooterButton(localize('cancel', "Cancel"), () => this.onCancelHandler());
+
+		this._register(attachInputBoxStyler(this._viewNameInput!, this._themeService));
+		this._register(attachButtonStyler(this._submitButton, this._themeService));
+		this._register(attachButtonStyler(this._cancelButton, this._themeService));
+
+		this._register(this._viewNameInput.onDidChange(v => this.validate()));
+
+		attachModalDialogStyler(this, this._themeService);
+		this.validate();
+	}
+
+	private validate() {
+		let valid = true;
+
+		if (!this._viewNameInput.validate()) {
+			valid = false;
+		}
+
+		this._submitButton.enabled = valid;
+	}
+
+	private onSubmitHandler() {
+		this._view.name = this._viewNameInput.value;
+		this._view.save();
+
+		this.close();
+	}
+
+	private onCancelHandler() {
+		this.close();
+	}
+
+	public close(): void {
+		return this.hide();
+	}
+
+	public open(): void {
+		this.show();
+	}
+
+	public override dispose(): void {
+		super.dispose();
+		for (let key in this._optionsMap) {
```

Comment:
You could also just register a disposable in the constructor that cleans up the options map at that point as well